device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -401,6 +401,17 @@ public void addMessage(Message message, IotHubEventCallback callback, Object cal
 
         //Codes_SRS_IOTHUBTRANSPORT_34_042: [This function shall build a transport packet from the provided message,
         // callback, and context and then add that packet to the waiting queue.]
+
+        if (message.isBulk() && !(this.iotHubTransportConnection instanceof HttpsIotHubConnection))
+        {
+            for (Message singleMessage : message.getNestedMessages())
+            {
+                waitingPacketsQueue.add(new IotHubTransportPacket(message, callback, callbackContext,null, System.currentTimeMillis()));
```

Comment:
Use ```singleMessage``` when constructing the iothubtransportpacket, not ```message```

Reply:
Otherwise you'll just be sending the bulk message over and over, not its children.